src/ripple/overlay/impl/OverlayImpl.h

```diff
@@ -456,6 +456,41 @@ class OverlayImpl : public Overlay
 
     void
     sendEndpoints();
+
+private:
+    struct Stats
+    {
+        template <class Handler>
+        Stats (Handler const& handler, beast::insight::Collector::ptr const& collector)
+            : hook (collector->make_hook (handler))
+            , bytesIn(collector->make_counter("Overlay","Bytes_In"))
+            , bytesOut(collector->make_counter("Overlay","Bytes_Out"))
+            , messagesIn(collector->make_counter("Overlay","Messages_In"))
+            , messagesOut(collector->make_counter("Overlay","Messages_Out"))
+            { }
+
+        beast::insight::Hook hook;
+        beast::insight::Counter bytesIn;
+        beast::insight::Counter bytesOut;
+        beast::insight::Counter messagesIn;
+        beast::insight::Counter messagesOut;
+    };
+
+    Stats m_stats;
+
+private:
+    void collect_metrics()
+    {   
+        std::lock_guard lock (mutex_);
+        auto const stats = m_traffic.getCounts();
+        for (auto const& i : stats)
+        {
```

Comment:
This PR has an issue: it never reports the _name_ of the counters. It merely coagulates all different categories into a single `struct`.

We need one set of counters for every message, which complicates things slighty. Consider changing `m_stats` to be `std::vector<Stats>` and adding a `name` to the `Stats` object.

Reply:
Thanks, I'm on it.

Reply:
@nbougalis I made the changes you requested.

Reply:
:vulcan_salute: